crates/escrow/src/tests.rs

```diff
@@ -223,3 +224,34 @@ fn deposit_below_max_height_truncates_to_zero() {
         );
     })
 }
+
+#[test]
+fn should_update_stake() {
+    run_test(|| {
+        let start_time = System::block_number();
+        let max_time = start_time + MaxPeriod::get();
+        let end_time = max_time;
+        let amount = 1000;
+
+        create_lock(BOB, amount, end_time);
+
+        let bob_initial_balance: Balance = <Rewards>::get_stake(&(), &BOB).unwrap();
+        assert_eq!(bob_initial_balance, 1000_u128.into());
+
+        System::set_block_number(50);
+
+        assert_ok!(Escrow::update_user_stake(RuntimeOrigin::signed(ALICE), BOB));
+
+        let bob_final_balance: Balance = <Rewards>::get_stake(&(), &BOB).unwrap();
+        assert_eq!(bob_final_balance, 500_u128.into());
```

Comment:
could you add a check that the accumulated rewards don't change?

Reply:
@sander2 are u referring to this [compute_reward](https://github.com/interlay/interbtc/blob/ccbdbac80835baaf2c0e112035506d9a8a31e3e4/crates/reward/src/lib.rs#L288) call to get rewards before and after calling the extrinsic.

Reply:
Yea that's it

Reply:
added extra assertions in test cases after going through the rewards pallet.